### PR TITLE
Add new quirks for SOL001 parsing

### PR DESCRIPTION
**Pull request by jpc4242**
The new quirks are:
- imports.ignore.topology_template: ignore topology_template in imported units
- substitution_mappings.requirements.allow_dangling: allow a requirement in substitution_mapping to point to an undeclared requirement of a node in node_template

### CODE REVIEW

**Review by tliron — CHANGES_REQUESTED** [review]

File: tosca/grammars/tosca_v1_1/unit.go [hunk 1]
```diff
@@ -18,6 +18,10 @@ func ReadUnit(context *tosca.Context) tosca.EntityPtr {
 
 	self := tosca_v2_0.NewUnit(context)
 	context.ScriptletNamespace.Merge(DefaultScriptletNamespace)
-	context.ValidateUnsupportedFields(append(context.ReadFields(self), "dsl_definitions"))
+        if context.HasQuirk(tosca.QuirkImportsIgnoreTopologyTemplate) {
```
[the remark this answers is not in the document]
Comment (jpc4242) [reply tosca/grammars/tosca_v1_1/unit.go]: Fixed in later commit

File: tosca/quirk.go [hunk 2]
```diff
@@ -44,10 +44,15 @@ const QuirkNamespaceNormativeShortcutsDisable Quirk = "namespace.normative.short
 // quirk changes the expected syntax to be a sequenced list.
 const QuirkSubstitutionMappingsRequirementsList Quirk = "substitution_mappings.requirements.list"
 
+// ETSI SOL001 2.x mandates usage of imports that contain topology_template
+// this quirk ignores topology_template in imports
 //
-// Quirks
-//
+const QuirkImportsIgnoreTopologyTemplate Quirk = "imports.ignore.topology_template"
```
[the remark this answers is not in the document]
Comment (jpc4242) [reply tosca/quirk.go]: Fixed in later commit

File: tosca/quirk.go [hunk 3]
```diff
@@ -44,10 +44,15 @@ const QuirkNamespaceNormativeShortcutsDisable Quirk = "namespace.normative.short
...
 
+// ETSI SOL001 2.x needs substitutionmappings using unassigned referenced
+const QuirkSubstitutionMappingsRequirementsAllowDangling Quirk = "substitution_mappings.requirements.allow_dangling"
```
[the remark this answers is not in the document]
Comment (jpc4242) [reply tosca/quirk.go:53]: I opened discussion at https://github.com/tliron/puccini/discussions/61

**Comment by tliron** [comment]
Hi! Sorry for taking so long to review. Really the only issue is to please squash it into one commit, then we will merge.

**Pull request closed without merging**